package/tests/test_integrations/test_run_hooks.py

```diff
@@ -0,0 +1,199 @@
+from __future__ import annotations
+
+import sys
+from types import ModuleType
+from typing import Any, Dict
+
+import pytest
+from kedro.io import DataCatalog, MemoryDataset
+from kedro.pipeline import Pipeline, node
+from kedro.pipeline.node import Node as KedroNode
+
+from kedro_viz.integrations.kedro.run_hooks import PipelineRunStatusHook
+
+
+# -----------------------------------------------------------------------------
+# Fixtures
+# -----------------------------------------------------------------------------
+@pytest.fixture()
+def hooks() -> PipelineRunStatusHook:
+    """Fresh hook instance for each test."""
+    return PipelineRunStatusHook()
+
+
+@pytest.fixture()
+def sample_node() -> KedroNode:
+    return node(lambda x: x, inputs="in", outputs="out", name="test_node")
+
+
+@pytest.fixture()
+def sample_pipeline(sample_node) -> Pipeline:
+    return Pipeline([sample_node])
+
+
+@pytest.fixture()
+def sample_catalog() -> DataCatalog:
+    """Minimal DataCatalog with a single in-memory dataset."""
+    return DataCatalog({"memory": MemoryDataset()})
+
+
+# -----------------------------------------------------------------------------
+# after_catalog_created
+# -----------------------------------------------------------------------------
+def test_after_catalog_created_standard(hooks, sample_catalog):
+    hooks.after_catalog_created(sample_catalog)
+    assert hooks._datasets == sample_catalog._datasets
+
+
+def test_after_catalog_created_import_error(hooks, sample_catalog, monkeypatch):
```

Comment:
I think we need to remove this one as well. because we are not going to be backward compatible. I think we keep it in the code for now; so until Kedro 1.0 is released our CI/CD passes. But let's remove tehse tests and add a note in the code to remove that fallback before we release.